### PR TITLE
JPA

### PR DESCRIPTION
**Pull request by DmitriyLitvin**

### CODE REVIEW

File: src/main/java/mate/academy/jpademo/Main.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package mate.academy.jpademo;
+
+import mate.academy.jpademo.model.Patient;
+import mate.academy.jpademo.model.TypeOfTest;
+import mate.academy.jpademo.model.device.Photometer;
+import mate.academy.jpademo.model.device.UltrasonicDevice;
+import mate.academy.jpademo.service.factory.ServiceFactory;
+
+import java.time.LocalDate;
+
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/Main.java:10]: line not need

File: src/main/java/mate/academy/jpademo/Main.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Photometer photometer = new Photometer();
+        photometer.setRadius(20.0);
+        photometer.setWeight(25.5);
+        photometer.setTransmittance(340);
+        ServiceFactory.getDeviseService().create(photometer);
+        ServiceFactory.getTestService().doTest(patient, TypeOfTest.BLOOD,photometer);
+        ServiceFactory.getTestService().getAllTests().stream().forEach(System.out::println);
+        ServiceFactory.getPatientService().getPatientsWithGlucoseHigherThan(130.0).stream().forEach(System.out::println);
+        ServiceFactory.getTestService().getAllTestsInDateRange(LocalDate.of(1997,10,10),LocalDate.now());
+
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/Main.java]: line not need

File: src/main/java/mate/academy/jpademo/model/Patient.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.time.LocalDate;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/model/Patient.java]: You don't use List

File: src/main/java/mate/academy/jpademo/model/accessory/UltrasonicAccessory.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.jpademo.model.accessory;
+
+import mate.academy.jpademo.model.device.Device;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/model/accessory/UltrasonicAccessory.java]: You don't use annotation "Table"

File: src/main/java/mate/academy/jpademo/service/PatientService.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package mate.academy.jpademo.service;
+
+import mate.academy.jpademo.model.Patient;
+
+import java.util.List;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/service/PatientService.java]: You don't use List

File: src/main/java/mate/academy/jpademo/service/TestService.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package mate.academy.jpademo.service;
+
+import mate.academy.jpademo.model.TypeOfTest;
+import mate.academy.jpademo.model.Patient;
+import mate.academy.jpademo.model.device.Device;
+import mate.academy.jpademo.model.test.Test;
+
+import java.time.LocalDate;
+import java.util.List;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/service/TestService.java]: You don't use List

File: src/main/java/mate/academy/jpademo/service/impl/PatientServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package mate.academy.jpademo.service.impl;
+
+import mate.academy.jpademo.dao.PatientDao;
+import mate.academy.jpademo.model.Patient;
+import mate.academy.jpademo.service.PatientService;
+
+import java.util.Set;
+
+public class PatientServiceImpl implements PatientService {
+    PatientDao patienceDao;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/service/impl/PatientServiceImpl.java]: private

File: src/main/java/mate/academy/jpademo/dao/impl/PatientDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+import mate.academy.jpademo.model.Patient;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.TypedQuery;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class PatientDaoImpl extends AbstractDao implements PatientDao {
+
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/dao/impl/PatientDaoImpl.java:14]: line not need

File: src/main/java/mate/academy/jpademo/model/test/SkinTest.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+import mate.academy.jpademo.model.device.Device;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import java.time.LocalDate;
+
+@Entity(name = "skin_test")
+public class SkinTest extends Test {
+    @Column
+    Double dryness;
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/jpademo/model/test/SkinTest.java]: private

**Review by BondarenkoJek — CHANGES_REQUESTED** [review]

**Review by BondarenkoJek — APPROVED** [review]
LGTM

**Review by boroda4436 — APPROVED** [review]
LGTM

**Pull request merged**